include/obmc_console.hpp

```diff
@@ -132,6 +133,67 @@ inline void requestRoutes(App& app)
                     conn.getIoContext());
             hostSocket->async_connect(ep, connectHandler);
         }
+        // Ensure user has ConfigureManager, setting above does nothing
+        auto getUserInfo =
+            [&conn](
+                const boost::system::error_code ec,
+                boost::container::flat_map<
+                    std::string,
+                    std::variant<bool, std::string, std::vector<std::string>>>
+                    userInfo) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR << "GetUserInfo failed...";
+                conn.close("Failed to get user information");
+                return;
+            }
+
+            const std::string* userRolePtr = nullptr;
+            auto userInfoIter = userInfo.find("UserPrivilege");
+            if (userInfoIter != userInfo.end())
+            {
+                userRolePtr = std::get_if<std::string>(&userInfoIter->second);
+            }
+
+            std::string userRole{};
+            if (userRolePtr != nullptr)
+            {
+                userRole = *userRolePtr;
+                BMCWEB_LOG_DEBUG << "userName = " << conn.getUserName()
+                                 << " userRole = " << *userRolePtr;
+            }
+
+            // Get the user privileges from the role
+            ::redfish::Privileges userPrivileges =
+                ::redfish::getUserPrivileges(userRole);
+
+            const ::redfish::Privileges requiredPrivileges{"ConfigureManager"};
+
+            if (!userPrivileges.isSupersetOf(requiredPrivileges))
+            {
+                BMCWEB_LOG_DEBUG
+                    << "User " << conn.getUserName()
+                    << " not authorized for host console connection";
+                conn.close("Unathourized access");
+                return;
+            }
+
+            sessions.insert(&conn);
+            if (hostSocket == nullptr)
+            {
+                const std::string consoleName("\0obmc-console", 13);
+                boost::asio::local::stream_protocol::endpoint ep(consoleName);
+
+                hostSocket = std::make_unique<
+                    boost::asio::local::stream_protocol::socket>(
+                    conn.getIoContext());
+                hostSocket->async_connect(ep, connectHandler);
+            }
+        };
+        crow::connections::systemBus->async_method_call(
```

Comment:
Any pre-req on user manager code in 1050 before this can go?

Reply:
I don't think so
but I haven't been able to test. See statement 
```
Tested: This worked, was tested on 1030 but for 1050. Even with and without this code hit

websocket console0/ closed.
            code: 1006
            reason: 

from the GUI
```

Reply:
This can wait too until we figure out what is up with the websockets for 1050